consensus/pbft/backends/simple/engine.go

```
@@ -210,18 +210,22 @@ func (sb *simpleBackend) RemovePeer(peerID string) error {
 func (sb *simpleBackend) HandleMsg(peerID string, data []byte) error {
 	peer := sb.peerSet.Get(peerID)
 	if peer == nil {
+		sb.logger.Error("Not in peer set", "peerID", peerID)
 		return nil
 	}
 
 	msgEvent, err := Decode(data)
 	if err != nil {
+		sb.logger.Error("Decode message event failed", "error", err)
 		return err
 	}
 
-	if val := sb.valSet.GetByAddress(peer.Address()); val != nil {
-		go sb.pbftEventMux.Post(*msgEvent)
+	if val := sb.valSet.GetByAddress(peer.Address()); val == nil {
+		sb.logger.Error("Not in validator set", "peerAddr", peer.Address().Hex())
+		return nil
```

Should we return an error in this case?